File: CSharp/XhtmlDomTest/XhtmlDom/Xhtml_Base.cs
using HtmlTest.Base;
using HtmlWalker.Platform;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.IO;
using XhtmlDomWalker;
using XhtmlXsd;

namespace HtmlTest.XhtmlDom.Base
{
    [TestClass]
    public class Xhtml_Base : CloneWalker_Base
    {
        public Xhtml_Base() : base("xhtml") { }

        protected override WalkerPlatform Platform => XhtmlDomPlatform.Instance;

        protected override void CreateExpected(FileInfo expected, FileInfo testInput, Options options = null)
        {
            try
            {
                var document = XhtmlEntityResolver.CreateXmlDocument();
                document.Load(testInput.FullName);
                document.Save(expected.FullName);
            }
            catch (Exception e)
            {
                PrintStackTrace(e);
                Assert.Fail(e.Message);
            }
        }

        [TestMethod]
        public override void TWalk()
        {
            base.TWalk();
        }

        [TestMethod]
        public override void TUnsupported()
        {
            base.TUnsupported();
        }

        [TestMethod]
        public override void TLoadHtml()
        {
            base.TLoadHtml();
        }

        [TestMethod]
        public override void TLoadPath()
        {
            base.TLoadPath();
        }

        [TestMethod]
        public override void TLoadHtmWalk()
        {
            base.TLoadHtmWalk();
        }

        [TestMethod]
        public override void TLoadPathWalk()
        {
            base.TLoadPathWalk();
        }

        [TestMethod]
        public override void TSavePath()
        {
            base.TSavePath();
        }

        [TestMethod]
        public override void TSaveTextWriter()
        {
            base.TSaveTextWriter();
        }

        [TestMethod]
        public override void TEntityIsPreserved()
        {
            base.TEntityIsPreserved();
        }

        [TestMethod]
        public override void TEntityIsPreservedSaved()
        {
            base.TEntityIsPreservedSaved();
        }

        [TestMethod]
        public override void TEntityIsPreservedLoadSaved()
        {
            base.TEntityIsPreservedLoadSaved();
        }

        [TestMethod]
        public override void TRemoveTag()
        {
            base.TRemoveTag();
        }
    }
}

File: Java/htmlwalker/src/test/java/htmltest/xhtmldom/Xhtml_Base.java
package htmltest.xhtmldom;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

import javax.xml.transform.TransformerException;
import javax.xml.transform.dom.DOMSource;
import javax.xml.transform.stream.StreamResult;

import org.junit.Test;
import org.w3c.dom.Document;
import org.xml.sax.SAXException;

import htmltest.base.CloneWalker_Base;
import htmlwalker.platform.WalkerPlatform;
import xhtmldomwalker.XhtmlDomDocument;
import xhtmldomwalker.XhtmlDomPlatform;

public class Xhtml_Base extends CloneWalker_Base
{
    public Xhtml_Base() { }

    @Override
    protected boolean isXml() { return true; }

    @Override
    protected WalkerPlatform platform() { return XhtmlDomPlatform.theInstance; }


	@Override
	protected void createExpected(File expected, File testInput, TestOptions options)
	{
		try {
			Document dom = XhtmlDomPlatform.documentBuilder().parse(testInput);
			DOMSource source = new DOMSource(dom);
			FileWriter writer = new FileWriter(expected.getAbsolutePath());
			StreamResult result = new StreamResult(writer);		
			XhtmlDomDocument.transformer().transform(source, result);
		}
		catch (IOException | SAXException | TransformerException e) {
			e.printStackTrace();
		}		
	}

    @Override
    @Test
	public void tConfirmTestFolder()
	{
        super.tConfirmTestFolder();
	}

    @Override
    @Test
	public void tConfirmOutFolder()
	{
        super.tConfirmOutFolder();
	}

    @Override
    @Test
    public void tWalk()
    {
        super.tWalk();
    }

    @Override
    @Test
    public void tUnsupported()
    {
        super.tUnsupported();
    }

    @Override
    @Test
    public void tLoadHtml()
    {
        super.tLoadHtml();
    }

    @Override
    @Test
    public void tLoadPath()
    {
        super.tLoadPath();
    }

    @Override
    @Test
    public void tLoadHtmWalk()
    {
        super.tLoadHtmWalk();
    }

    @Override
    @Test
    public void tLoadPathWalk()
    {
        super.tLoadPathWalk();
    }

    @Override
    @Test
    public void tSavePath()
    {
        super.tSavePath();
    }

    @Override
    @Test
    public void tSaveWriter()
    {
        super.tSaveWriter();
    }

    @Test
    public void tEntityIsPreserved()
    {
    	// TODO: Get entities working
    	//super.tEntityIsPreserved();
    }

    @Test
    public void tEntityIsPreservedSaved()
    {
    	// TODO: Get entities working
    	//super.tEntityIsPreservedSaved();
    }

    @Test
    public void tEntityIsPreservedLoadSaved()
    {
        super.tEntityIsPreservedLoadSaved();
    }
}
